rpcserver.go

```diff
@@ -1943,15 +1980,14 @@ func (r *rpcServer) parseOpenChannelReq(in *lnrpc.OpenChannelRequest,
 	globalFeatureSet := r.server.featureMgr.Get(feature.SetNodeAnn)
 
 	// Ensure that a local funding amount has been specified.
-	if localFundingAmt == 0 {
+	if localFundingAmt == 0 && !in.FundMax {
 		return nil, fmt.Errorf("local funding amount must be non-zero")
 	}
 
 	// Ensure that the initial balance of the remote party (if pushing
 	// satoshis) does not exceed the amount the local party has requested
 	// for funding.
-	//
-	if remoteInitialBalance >= localFundingAmt {
+	if remoteInitialBalance >= localFundingAmt && !in.FundMax {
```

Comment:
this also needs to be checked when fundMaxAmt is smaller then fundMinAmt => maybe where the final max amount is determined ?